app/models/metric/ci_mixin/processing.rb

```diff
@@ -105,6 +105,9 @@ def perf_process(interval_name, start_time, end_time, counters_data)
   private
 
   def transform_resources!(counters_data)
+    # no work needed if keys are already ActiveRecord objects
+    return if counters_data.first.first.kind_of?(ActiveRecord::Base)
```

Comment:
`counters_data` is a hash right and you're checking the first key?  Wonder if `counters_data.keys.first` would be more obvious as to what we're actually doing here

Reply:
Do we need to check `keys.all? { |k| k.kind_of?(ActiveRecord::Base) }` or is it not possible to have mixed types?

Reply:
yea. `first.first` is more efficient than `keys.first` but I think readability is best for us, especially since we are limiting this to 250 objects (for vmware at least)

Reply:
updated: I couldn't get this code the way I wanted it.
Decided to delete instead